docs/docs/configuration/pages.md

```diff
@@ -95,7 +97,17 @@ export default function SignIn({ providers }) {
 }
 
 export async function getServerSideProps(context) {
-  const providers = await getProviders()
+  const session = await getServerSession(context.req, context.res, authOptions);
+  
+  // If sign in is successful, redirect to homepage
+  if (session) {
+    return {
+      redirect: { destination: "/" },
+    };
+  }
```

Comment:
```suggestion
  // If the user is already logged in, redirect,
  // unless the destination is on the same page,
  // to prevent infinite redirects
  const destination = "/"
  if (session && destination !== authOptions.pages?.signIn) {
    return { redirect: { destination } };
  }
```